com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    private List<CloudServiceExtended> getExistingServices(List<CloudServiceExtended> servicesToProcess,
+        CloudControllerClient controllerClient) {
+        return servicesToProcess.parallelStream()
+            .map(service -> controllerClient.getService(service.getName(), false))
+            .filter(Objects::nonNull)
+            .map(service -> ImmutableCloudServiceExtended.builder()
+                .from(service)
+                .build())
+            .collect(Collectors.toList());
+    }
+
+    private Map<CloudServiceExtended, ServiceOperation> getServicesOperationInProgress(List<CloudServiceExtended> existingServices,
+        CloudControllerClient controllerClient, String spaceId) {
+        Map<CloudServiceExtended, ServiceOperation> servicesOperation = new HashMap<>();
+        existingServices.forEach(service -> {
+            Map<String, Object> serviceInstanceEntity = getServiceInstanceEntity(controllerClient, service, spaceId);
+            ServiceOperation serviceOperationInProgress = getServiceOperationInProgress(serviceInstanceEntity);
+            if (serviceOperationInProgress != null) {
+                servicesOperation.put(service, serviceOperationInProgress);
+            }
+        });
+        return servicesOperation;
+    }
+
+    private Map<String, Object> getServiceInstanceEntity(CloudControllerClient client, CloudService service, String spaceId) {
+        return serviceInstanceGetter.getServiceInstanceEntity(client, service.getName(), spaceId);
+    }
+
+    private ServiceOperation getServiceOperationInProgress(Map<String, Object> serviceInstanceEntity) {
+        ServiceOperation lastOpertaion = getLastOperation(serviceInstanceEntity);
+        if (lastOpertaion != null && lastOpertaion.getState() == ServiceOperationState.IN_PROGRESS) {
+            return lastOpertaion;
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    private ServiceOperation getLastOperation(Map<String, Object> cloudServiceInstance) {
+        Map<String, Object> lastOperationAsMap = (Map<String, Object>) MapUtils.getMap(cloudServiceInstance, LAST_SERVICE_OPERATION);
+        if (lastOperationAsMap == null) {
+            return null;
+        }
+        return parseServiceOperationFromMap(lastOperationAsMap);
+    }
+
+    private ServiceOperation parseServiceOperationFromMap(Map<String, Object> serviceOperation) {
+        if (serviceOperation.get(SERVICE_OPERATION_TYPE) == null || serviceOperation.get(SERVICE_OPERATION_STATE) == null) {
+            return null;
+        }
+        ServiceOperationType type = ServiceOperationType.fromString((String) serviceOperation.get(SERVICE_OPERATION_TYPE));
+        ServiceOperationState state = ServiceOperationState.fromString((String) serviceOperation.get(SERVICE_OPERATION_STATE));
+        String description = (String) serviceOperation.get(SERVICE_OPERATION_DESCRIPTION);
+        return new ServiceOperation(type, description, state);
```

Comment:
extract this:
` ServiceOperationType type = ServiceOperationType.fromString((String) serviceOperation.get(SERVICE_OPERATION_TYPE));
         ServiceOperationState state = ServiceOperationState.fromString((String) serviceOperation.get(SERVICE_OPERATION_STATE));
         String description = (String) serviceOperation.get(SERVICE_OPERATION_DESCRIPTION);
         return new ServiceOperation(type, description, state);`
to another method - mapToServiceOperation